src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializer.java

```diff
@@ -302,20 +312,32 @@ public Pair<KudoHostMergeResult, MergeMetrics> mergeOnHost(List<KudoTable> kudoT
               metricsBuilder::mergeIntoHostBufferTime);
 
     return Pair.of(result, metricsBuilder.build());
-
   }
 
   /**
-   * Merge a list of kudo tables into a contiguous table.
+   * Merge an array of kudo tables into a contiguous table.
    * <br/>
    * The caller should ensure that the {@link KudoSerializer} used to generate kudo tables have same schema as current
    * {@link KudoSerializer}, otherwise behavior is undefined.
    *
-   * @param kudoTables list of kudo tables. This method doesn't take ownership of the input tables, and caller should
+   * @param kudoTables array of kudo tables. This method doesn't take ownership of the input tables, and caller should
    *                   take care of closing them after calling this method.
-   * @return the merged table, and metrics during merge.
+   * @return the merged table.
    * @throws Exception if any error occurs during merge.
    */
+  public Table mergeToTable(KudoTable[] kudoTables) throws Exception {
+    try (KudoHostMergeResult children = mergeOnHost(kudoTables)) {
+      return children.toTable();
+    }
+  }
+
+
+  /**
+   * See {@link #mergeToTable(KudoTable[])}.
+   *
+   * @deprecated Use {@link #mergeToTable(KudoTable[])} instead.
+   */
+  @Deprecated
```

Comment:
are we planning to delete this in another PR ?

Reply:
Is it mandatory to approach it in such a tedious way?  Can we just make changes at both spark-rapids-jni side and spark-rapids side, then merge the PR simultaneously? The reason why I ask this is that I also have some incompatible (so I cannot use this Deprecate annotation way) PR changes in spark-rapids-jni at hand, and I'm wondering what's the best way to check in such kind of changes, also cc @revans2